### PR TITLE
Qt/Loader: Let users choose which packages to install

### LINKED ISSUE #12177: [Feature Request] Installation dialogue for games installed from pkgdir
## Quick summary
Some discs have pkgs on the disc inside PS3_GAME/PKGDIR

## Details
The current behavior is RPCS3 just says (not responding) during first boot of Bioshock infinite while installing hdd game Bioshock (included on disc) and during first boot of Assassin's Creed Revelations installing hdd game Assassin's Creed (included on disc).

There is no user feedback that anything is happening at all so we're looking at the screen like what is happening?
For those knowledgeable by having been informed about the issue previously know that all that is happening is you're installing a hdd game from your disc so give it few minutes.

We should probably include an option to not install the hdd game upon users request and they can tick "don't show this message again" for those who just don't want to install the game.
If they did want to install it they can always manually drag the PKG onto RPCS3's UI to install the pkg.

Current behavior forces users to install the hdd game without permission and sometimes without the user's knowledge.

#### 1. Please describe, what part of RPCS3 would be affected by your feature:
* UI
* Game Installation

#### 2. Please tell us, why your feature is important to RPCS3.
To inform users that RPCS3 is working as intended.

#### 3. Please attach screenshots of the feature implemented in other projects.

I'm not aware of any.
We just need a qmessagebox or something to reflect information to the user.

#### 4. Please provide your system configuration:
* Win11
* R5 3600
* GTX 1660S

#### Please include.
* Anything else you deem to be important

If you want to throw in detecting and installing dlc pkgs from the subdirectories of the PS3_EXTRA folder of games like skyrim to install DLC from disc at first boot like how is already implemented for hdd games installing from pkgdir it wouldn't hurt anything.

### PR DESCRIPTION
**Pull request by Megamouse**
Fixes #12177

### CODE REVIEW

**Comment by Darkhost1999** [comment]
Booting AC America's collection gives the batch install prompt 🥇 

image

Accept warns about closing your current game. Installs the content and gives success message.
Potentially unrelated but documenting my experience I left the create shortcut buttons empty and clicked accept "thank you for installing successfully" and got this vm:alloc error.

image
[RPCS3.log](https://github.com/RPCS3/rpcs3/files/11620067/RPCS3.log)

Next up we have Skyrim Legendary edition with the PS3_EXTRA folder and unfortunately just like master the entire folder was completely ignored and no DLC installed

image
[RPCS3.log](https://github.com/RPCS3/rpcs3/files/11620244/RPCS3.log)

I don't have any games with INSDIR but I did need to try a game with 1 pkg file instead of more than 1 and what if I said No to the digital manual of TLOU and so the single pkg dialog accepted it's rejection like a big boy and didn't ask me again to install what I told it not to.
[RPCS3.log](https://github.com/RPCS3/rpcs3/files/11620255/RPCS3.log)

This is a real game changer for communicating with the users whats going on. I really appreciate this work and love what I see.

**Comment by Darkhost1999** [comment]
[RPCS3.log](https://github.com/RPCS3/rpcs3/files/11620337/RPCS3.log)
The vm::alloc(): Invalid memory location (0) seems to be a recurring error
[RPCS3.log](https://github.com/RPCS3/rpcs3/files/11620348/RPCS3.log)
And occurs with different games.

**Comment by elad335** [comment]
@Darkhost1999  Test Skrim with #13934

**Comment by Megamouse** [comment]
I forgot to set some parameter while half asleep. Please try the AC stuff again.

File: rpcs3/rpcs3qt/main_window.cpp [hunk 1]
```diff
@@ -973,7 +990,9 @@ void main_window::HandlePackageInstallation(QStringList file_paths)
 		}
 	}
 
-	if (worker())
```
Comment (elad335) [inline rpcs3/rpcs3qt/main_window.cpp:976]: You do not need a value keeper, if you call worker.operator()() again it would yeild the same result.

Comment (Megamouse) [reply rpcs3/rpcs3qt/main_window.cpp:976]: The motivation here was that the code was not particularly self explanatory.
Having a variable telling you what this worker call means makes the code much easier to read

**Comment by Darkhost1999** [comment]
Cool with the message the game will be closed if you install packages I didn't think the game was continuing itself after install
[RPCS3.log](https://github.com/RPCS3/rpcs3/files/11622685/RPCS3.log)
But that does explain the error that's fixed. 
[RPCS3.log](https://github.com/RPCS3/rpcs3/files/11622714/RPCS3.log)
Skyrim easily listed the dlc names and languages for easy install and tlou works too accept an decline to install package.

I don't have Insdir to test that foldee and functionality

**Comment by elad335** [comment]
It may be best for it to execute on a PPU or other emulation thread rather than blocking the main thread.

**Comment by Megamouse** [comment]
I don't see a reason for that.
We want to block the main thread while the user is doing potentially weird things.

**Comment by elad335** [comment]
You make an assumption that all signals prior to an X blocking operation are erased and are not invoked. But it is not true in Qt. It's of course not the only place this happens so I may create an issue later.

**Pull request merged**